client-tez/src/main/java/org/apache/tez/common/RssTezUtils.java

```diff
@@ -0,0 +1,451 @@
+            RssTezConfig.RSS_RUNTIME_IO_SORT_MB + initialMemRequestMb
+                    + " should be " + "larger than 0 and should be less than the available task memory (MB):"
+                + (maxAvailableTaskMemory >> 20));
+    LOG.info("Requested BufferSize (" + TezRuntimeConfiguration.TEZ_RUNTIME_UNORDERED_OUTPUT_BUFFER_SIZE_MB
+            + ") : " + initialMemRequestMb);
+    return reqBytes;
+  }
+
+  public static String uniformPartitionHostInfo(Map<Integer, List<ShuffleServerInfo>> map) {
+    List<String> res = new ArrayList<>();
+    String tmp;
+    Set<Integer> pidSet = map.keySet();
+    for (Integer pid : pidSet) {
+      for (ShuffleServerInfo shuffleServerInfo : map.get(pid)) {
+        tmp = pid + UNDERLINE_DELIMITER + shuffleServerInfo.getHost() + COLON_DELIMITER + shuffleServerInfo.getNettyPort();
+        res.add(tmp);
+      }
+    }
+    return org.apache.commons.lang.StringUtils.join(res, COMMA_DELIMITER);
+  }
+
+  public static Map<String, List<String>> uniformServerToPartitions(String partitionToServers) {
+    Map<String, List<String>> serverToPartitions = new HashMap<>();
+    List<String> list;
+
+    String[] pidWithWorkerInfos = partitionToServers.split(COMMA_DELIMITER);
+    for (String pidWithWorkerInfo : pidWithWorkerInfos) {
+      String[] pidUnderLineWorkerInfo = pidWithWorkerInfo.split(UNDERLINE_DELIMITER);
+      if (serverToPartitions.containsKey(pidUnderLineWorkerInfo[1])) {
+        list = serverToPartitions.get(pidUnderLineWorkerInfo[1]);
+        list.add(pidUnderLineWorkerInfo[0]);
+      } else {
+        list = new ArrayList<>();
+        list.add(pidUnderLineWorkerInfo[0]);
+        serverToPartitions.put(pidUnderLineWorkerInfo[1], list);
+      }
+    }
+
+    return serverToPartitions;
+  }
+
+  public static String uniformServerToPartitions(Map<String, List<String>> map) {
+    List<String> res = new ArrayList<>();
+    Set<String> keySet = map.keySet();
+    for (String s : keySet) {
+      String join = org.apache.commons.lang.StringUtils.join(map.get(s), UNDERLINE_DELIMITER);
+      res.add(s + PLUS_DELIMITER + join);
+    }
+
+    return org.apache.commons.lang.StringUtils.join(res,COMMA_DELIMITER);
+  }
+
+  public static ApplicationAttemptId getApplicationAttemptId() {
+    String containerIdStr = System.getenv(ApplicationConstants.Environment.CONTAINER_ID.name());
+    ContainerId containerId = ContainerId.fromString(containerIdStr);
+    return containerId.getApplicationAttemptId();
+  }
+
+  public static String uniqueIdentifierToAttemptId(String uniqueIdentifier) {
+    if (uniqueIdentifier == null) {
+      throw new RuntimeException("uniqueIdentifier should not be null");
+    }
+    String[] ids = uniqueIdentifier.split("_");
+    return StringUtils.join(ids, "_", 0, 7);
+  }
+
+  public static int getInt(Configuration rssJobConf, Configuration mrJobCOnf, String key, int defaultValue) {
```

Comment:
1. name fix
rssJobConf -> rssConf, mrJobCOnf -> tezConf
2. In client-mr mode, we use rss_conf.xml to transmit rss config. In client-tez mode, I think rss_conf.xml is unnecessary, so we can retain only one config in parameters.

Reply:
ok, we removeed the getInt method()

Reply:
ok, we removed the getInt method()